File: personal_project_back-end/common-service/src/main/java/com/example/common/service/models/ServiceModel.java
package com.example.common.service.models;
import com.example.common.service.enums.ServiceRoomStatusEnum;
import com.example.common.service.enums.ServiceTypeEnum;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.*;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ServiceModel extends BaseModel {
    @NotBlank(message = "Service name is required")
    private String name;

    @NotNull(message = "Price is required")
    @Positive(message = "Price must be greater than zero")
    private Long price;
    private Long quantity;
    private Boolean isActive;
    private ServiceRoomStatusEnum status;

    ServiceTypeEnum serviceType;
}

File: personal_project_front-end/src/models/ServiceModel.ts
import { ServiceRoomStatusEnum } from "@/enums/ServiceRoomStatusEnum";
import { ServiceTypeEnum } from "@/enums/ServiceTypeEnum";

export type ServiceModel = {
  id?: number;
  name: string;
  price: number;
  quantity?: number;
  isActive?: boolean;
  status?: ServiceRoomStatusEnum;
  serviceType?: ServiceTypeEnum;
};

export const defaultServiceModel: ServiceModel = {
  name: "",
  price: 0,
  quantity: 1,
  isActive: true,
  status: ServiceRoomStatusEnum.ACTIVE,
  serviceType: ServiceTypeEnum.OTHER,
};
